wristband/__init__.py

```diff
@@ -21,6 +22,62 @@ def wrapper(cls):
 api.route = types.MethodType(api_route, api)
 
 
+def get_all_releases():
+    data = requests.get('{}/apps'.format(releases_endpoint)).json()
+    return data
+
+
+def get_all_releases_of_app_in_env(deploy_env, app_name, releases):
+    releases_for_env = []
+    for release in releases:
+        if deploy_env == release.get("env") and app_name == release['an']:
+            del release['an']
+            del release['env']
+            releases_for_env.append(release)
+    return sorted(releases_for_env, key=lambda k: k['ls'], reverse=True)
+
+
+def get_all_app_names(releases):
+    all_app_names = sorted(set([release["an"] for release in releases]))
+    return all_app_names
+
+
+def get_all_app_names_in_env(env, releases):
+    apps_in_environment = sorted(set([release["an"] for release in releases if release.get("env") == env]))
+    return apps_in_environment
+
+
+def get_all_environments():
+    pipeline_list = get_all_pipelines()
+    environments = []
+    for pipeline in pipeline_list:
+        environments.extend(get_envs_in_pipeline(pipeline))
+    return sorted(environments)
+
+
+def get_all_pipelines():
+    return app.config.get('PIPELINES')
+
+
+def get_envs_in_pipeline(pipeline):
+    return app.config.get('PIPELINES')[pipeline]
+
+
+def make_environment_groups(environments):
+    shortenvs = sorted(set([short.split('-')[0] for short in environments]))
+    envgroups = {}
+    for shortenv in shortenvs:
+        envgroups[shortenv] = [env for env in environments if shortenv in env]
+    return envgroups
+
+
+def sse(event, data):
```

Comment:
this can be further factorised by passing a different data structure (tuples of key, data tuples) and passing a list comprehension into the join